analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsCatchExceptions.cs

```diff
@@ -0,0 +1,437 @@
+    {
+        if (arg == 42)
+        {
+            try
+            {
+                DoSomething();
+            }
+            catch
+            {
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void NestedCatch_Multi(int arg)  // Compliant
+    {
+        if (arg == 42)
+        {
+            try
+            {
+                DoSomething();
+            }
+            catch
+            {
+            }
+        }
+        else
+        {
+            try
+            {
+                DoSomething();
+            }
+            catch
+            {
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void NestedCatch_Deep(int arg)  // Compliant
+    {
+        if (arg == 42)
+        {
+            if (arg != 0)
+            {
+                while (true)
+                {
+                    try
+                    {
+                        DoSomething();
+                    }
+                    catch
+                    {
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void InvocationOutside_BeforeTry(int arg)   // Noncompliant
+    {
+        if (arg == 42)
+        {
+            DoSomething();
+            try
+            {
+                DoSomething();
+            }
+            finally
+            {
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void InvocationOutside_BeforeIf(int arg)   // Noncompliant
+    {
+        DoSomething();
+        if (arg == 42)
+        {
+            try
+            {
+                DoSomething();
+            }
+            finally
+            {
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void InvocationOutside_AfterTry(int arg)   // Noncompliant
+    {
+        if (arg == 42)
+        {
+            try
+            {
+                DoSomething();
+            }
+            finally
+            {
+            }
+            DoSomething();
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void InvocationOutside_AfterIf(int arg)   // Noncompliant
+    {
+        if (arg == 42)
+        {
+            try
+            {
+                DoSomething();
+            }
+            finally
+            {
+            }
+        }
+        DoSomething();
+    }
+
+    [FunctionName("Sample")]
+    public static void InvocationOutside_NestedDeep(int arg)  // Noncompliant
+    {
+        if (arg == 42)
+        {
+            if (arg != 0)
+            {
+                while (true)
+                {
+                    try
+                    {
+                        DoSomething();
+                    }
+                    catch
+                    {
+                    }
+                    DoSomething();
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
Some nested try test cases like this one are missing. And also try/catch in a catch or finally block 
```cs
try {
  DoSomething()
  try { }
  catch {}
} catch (InvalidOperationException) { }
```